### PR TITLE
K8SPG-91 Configurable SecurityContext

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

File: percona/controllers/pgc/pgc.go [hunk 1]
```diff
@@ -40,6 +42,7 @@ type Controller struct {
 const (
 	deploymentTemplateName = "cluster-deployment.json"
 	templatePath           = "/"
+	defaultSecurityContext = `{"fsGroup": 1001,"supplementalGroups": [1001, 1002, 1003]}`
```
Comment (hors) [inline percona/controllers/pgc/pgc.go:45]: pgo uses fsGroup 26 by default and all containers use 26

**Review by hors — CHANGES_REQUESTED** [review]

File: percona/controllers/deployment/deployment.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package deployment
+
+import (
+	crv1 "github.com/percona/percona-postgresql-operator/pkg/apis/crunchydata.com/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+func UpdateSpecTemplaeSpecSecurityContext(cl *crv1.PerconaPGCluster, deployment *appsv1.Deployment) {
```
Comment (qJkee) [inline percona/controllers/deployment/deployment.go:9]:
```suggestion
func UpdateSpecTemplateSpecSecurityContext(cl *crv1.PerconaPGCluster, deployment *appsv1.Deployment) {
```

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| clone-cluster | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/99/commits/5e494e13fb84ab180a10bf4d84a04ad7cc2bab6b
image: `perconalab/percona-postgresql-operator:PR-99-5e494e13-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-99-5e494e13-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-99-5e494e13-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-99-5e494e13-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-99-5e494e13-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-99-5e494e13-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| clone-cluster | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/99/commits/7fccabf3a17a025d11d08488dd7cf7abb54dfc31
image: `perconalab/percona-postgresql-operator:PR-99-7fccabf3-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-99-7fccabf3-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-99-7fccabf3-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-99-7fccabf3-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-99-7fccabf3-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-99-7fccabf3-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| clone-cluster | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/99/commits/991f1ce493b40c1eaedb2e7b40572c72720d11e7
image: `perconalab/percona-postgresql-operator:PR-99-991f1ce4-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-99-991f1ce4-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-99-991f1ce4-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-99-991f1ce4-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-99-991f1ce4-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-99-991f1ce4-pgo-deployer`

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| clone-cluster | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/99/commits/bec2d96756fded4659e9ae45be461ee3bbe2b630
image: `perconalab/percona-postgresql-operator:PR-99-bec2d967-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-99-bec2d967-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-99-bec2d967-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-99-bec2d967-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-99-bec2d967-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-99-bec2d967-pgo-deployer`

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| clone-cluster | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/99/commits/fce487f6d0dd4b0587ac347a8e2799eadc4b0c84
image: `perconalab/percona-postgresql-operator:PR-99-fce487f6-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-99-fce487f6-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-99-fce487f6-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-99-fce487f6-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-99-fce487f6-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-99-fce487f6-pgo-deployer`

**Pull request merged**